corehq/motech/openmrs/repeater_helpers.py

```diff
@@ -113,6 +115,42 @@ def url(url_format_string, **kwargs):
     return url_format_string.format(**kwargs)
 
 
+def get_case_location(case):
+    """
+    If the owner of the case is a location, return it. Otherwise return
+    the owner's primary location. If the case owner does not have a
+    primary location, return None.
+    """
+    case_owner = get_wrapped_owner(get_owner_id(case))
+    if isinstance(case_owner, SQLLocation):
+        return case_owner
+    location_id = case_owner.get_location_id(case.domain)
+    return SQLLocation.by_location_id(location_id) if location_id else None
```

Comment:
If the case owner is a user, you should be able to use `case_owner.get_location(case.domain)` (which may return `None`).  If the owner is a group, neither of those methods will be available, and I think this'll throw an error.